api/v1alpha1/hostedclusterinfrastructure_types.go

```diff
@@ -0,0 +1,344 @@
+
+	// Azure defines azure specific settings
+	Azure *AzurePlatformSpec `json:"azure,omitempty"`
+}
+
+// AWSPlatformInfraConfigStatus specifies configuration for clusters running on Amazon Web Services.
+type AWSPlatformInfraConfigStatus struct {
+
+	// VPC is the VPC to use for control plane cloud resources.
+	// HostedCluster.spec.platform.aws.cloudProviderConfig.vpc
+	//
+	// +optional
+	// +immutable
+	VPC string `json:"vpc,omitempty"`
+
+	// InstanceProfile is the AWS EC2 instance profile, which is a container for an IAM role that the EC2 instance uses.
+	InstanceProfile string `json:"instanceProfile,omitempty"`
+
+	// RolesRef contains references to various AWS IAM roles required to enable
+	// integrations such as OIDC.
+	// HostedCluster.spec.platform.aws.rolesRef
+	//
+	// +optional
+	// +immutable
+	RolesRef *AWSRolesRef `json:"rolesRef,omitempty"`
+
+	// SecurityGroups is an optional set of security groups to associate with node
+	// instances. One of more of the security groups can be used with nodePool resources
+	// NodePool.spec.platform.aws.securityGroups[]
+	//
+	// +optional
+	SecurityGroups []AWSResourceReference `json:"securityGroups,omitempty"`
+
+	// Zones are availability zones in an AWS region.
+	// An AWS subnet is created in each zone. The info is then used to populate
+	// HostedCluster.spec.platform.aws.cloudProviderConfig.zone
+	// HostedCluster.spec.platform.aws.cloudProviderConfig.subnet
+	// NodePool.spec.platform.aws.subnet.id
+	//
+	// +optional
+	Zones []AWSZoneAndSubnetConfigStatus `json:"zones,omitempty"`
+
+	// IndirectResources are created as part of the enablement for HostedCluster and NodePool(s). These indirect resources
+	// are associated with other AWS resources that are directly consumed by HostedCluster and NodePool(s).
+	IndirectResources AWSPlatformInfraIndirectConfigStatus `json:"indirectResources"`
+}
+
+type AWSZoneAndSubnetConfigStatus struct {
+	// Subnet will be created if value is empty in the specified zone
+	//
+	Subnet *AWSResourceReference `json:"subnet,omitempty"`
+
+	// PublicSubnet will be created for each zone that a private "Subnet" is created
+	//
+	PublicSubnet *AWSResourceReference `json:"publicSubnet,omitempty"`
+
+	// Zone is the availability zone to be used, a subnet will be created if one is not provided.
+	// The availability zones must be a memeber of the spec.platform.aws.region
+	//
+	Zone string `json:"zone"`
+}
+
+// HostedClusterInfrastructure defines the observed state of HostedClusterInfrastructure
+type HostedClusterInfrastructureStatus struct {
+	// Track the conditions for each step in the desired curation that is being
+	// executed as a job
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// Infrastructure is populated with the created or discovered infrastructure resouce IDs that are
+	// used when creating HostedCluster and  NodePool
+	Infrastructure HostedClusterInfrastructureConfigStatus `json:"infrastructure,omitempty"`
```

Comment:
why do we need this nested level? I'd expect HostedClusterInfrastructureStatus to have all the fields of `HostedClusterInfrastructureConfigStatus` and remove the config sufix everywhere?

Reply:
Could use something like `infrastructureStatus` to match the json name.

Reply:
But the nesting makes reading the status much easier.